src/Spark.Engine.Test/Service/PatchApplicationServiceTests.cs

```diff
@@ -0,0 +1,384 @@
+﻿namespace Spark.Engine.Test.Service
+{
+    using System.Linq;
+    using System.Reflection;
+    using Engine.Extensions;
+    using Engine.Service.FhirServiceExtensions;
+    using Hl7.Fhir.Model;
+    using Hl7.Fhir.Serialization;
+    using Xunit;
+    using Task = System.Threading.Tasks.Task;
+
+    public class PatchApplicationServiceTests
+    {
+        [Fact]
+        public void CanApplyPropertyAssignmentPatch()
+        {
+            var xml = @"<Parameters xmlns=""http://hl7.org/fhir"">
+  <parameter>
+    <name value=""operation""/>
+    <part>
+      <name value=""type""/>
+      <valueCode value=""add""/>
+    </part>
+    <part>
+      <name value=""path""/>
+      <valueString value=""Patient""/>
+    </part>
+    <part>
+      <name value=""name""/>
+      <valueString value=""birthDate""/>
+    </part>
+    <part>
+      <name value=""value""/>
+      <valueDate value=""1930-01-01""/>
+    </part>
+  </parameter>
+</Parameters>";
```

Comment:
Can you put all these in-line resources in the file system. There is a folder called TestData, and under TestData there is a folder for each FHIR Version R3, R4 and so on.

Reply:
Moved. For good measure, I implemented this feature in R4 because the PATCH verb is not exposed in previous versions.